x-pack/plugins/serverless_search/public/application/components/connectors/edit_service_type.tsx

```diff
@@ -18,41 +36,42 @@ interface EditServiceTypeProps {
   connector: Connector;
   isDisabled?: boolean;
 }
+interface OptionData {
+  secondaryContent?: string;
+}
+
+// TODO move these constants to a common file in Serverless as we have in Stack
+const BETA_LABEL = i18n.translate('xpack.serverlessSearch.betaLabel', {
+  defaultMessage: 'Beta',
+});
+
+const TECH_PREVIEW_LABEL = i18n.translate('xpack.serverlessSearch.techPreviewLabel', {
+  defaultMessage: 'Tech preview',
+});
 
 export const EditServiceType: React.FC<EditServiceTypeProps> = ({ connector, isDisabled }) => {
   const { http } = useKibanaServices();
   const connectorTypes = useConnectorTypes();
   const queryClient = useQueryClient();
   const { queryKey } = useConnector(connector.id);
+  const assetBasePath = useAssetBasePath();
 
-  const options =
-    connectorTypes.map((connectorType) => ({
-      inputDisplay: (
-        <EuiFlexGroup direction="row" alignItems="center">
-          <EuiFlexItem
-            grow={false}
-            data-test-subj={`serverlessSearchConnectorServiceType-${connectorType.serviceType}`}
-          >
-            <EuiIcon
-              size="l"
-              title={connectorType.name}
-              id={connectorType.serviceType}
-              type={connectorType.iconPath}
-            />
-          </EuiFlexItem>
-          <EuiFlexItem>{connectorType.name}</EuiFlexItem>
-        </EuiFlexGroup>
-      ),
-      value: connectorType.serviceType,
-    })) || [];
+  const allConnectors = useMemo(
+    () => connectorTypes.sort((a, b) => a.name.localeCompare(b.name)),
+    [connectorTypes]
+  );
 
   const { isLoading, mutate } = useMutation({
     mutationFn: async (inputServiceType: string) => {
       const body = { service_type: inputServiceType };
-      await http.post(`/internal/serverless_search/connectors/${connector.id}/service_type`, {
-        body: JSON.stringify(body),
-      });
-      return inputServiceType;
+      if (connector.service_type === null || connector.service_type === '') {
```

Comment:
What's the issue? If you are not calling the API you should not be getting the error anymore

Reply:
I'm still getting the described issue: 
https://github.com/user-attachments/assets/5cdd26bd-cd3d-4d39-bc50-aae78d226c67

Reply:
As discusses with @jedrazb for now we are preventing to make the API call if the `inputServiceType` is empty but it doesn't fix the issue to let users choose a _Customized connector_

```
const { isLoading, mutate } = useMutation({
    mutationFn: async (inputServiceType: string) => {
      const body = { service_type: inputServiceType };
      if (inputServiceType === null || inputServiceType === '') {
        return inputServiceType;
      } else {
        await http.post(`/internal/serverless_search/connectors/${connector.id}/service_type`, {
          body: JSON.stringify(body),
        });
        return inputServiceType;
      }
    },
    onSuccess: (successData) => {
      queryClient.setQueryData(queryKey, {
        connector: { ...connector, service_type: successData },
      });
      queryClient.invalidateQueries(queryKey);
    },
  });
```

Reply:
What do you mean with "it doesn't fix the issue"?

Reply:
Actually the same issue is in the current Serverless production version:

![CleanShot 2024-12-02 at 10 50 51](https://github.com/user-attachments/assets/c737279b-7ae1-4d2a-b210-c5bcee7e00c3)

Reply:
Here is the ticket to track this issue https://github.com/elastic/search-team/issues/8821

Reply:
> What do you mean with "it doesn't fix the issue"?

In the stack, when creating a customized connector, we leave the `service_type` field unpopulated.

Similarly, in ES3, for a customized connector, we don’t have a value to populate the service_type field in the connector document via the API, so we omit it from the update service type request. The side effect is that the combo box cannot distinguish between an empty state and a “customized connector state.” It might be worth setting a dummy placeholder service type, which would eventually be overridden by the connector framework—this needs to be verified. 

I think we can take it out of scope of this PR to merge what we have now